### PR TITLE
Move exec syscall filtering to NativeAccess

### PR DESCRIPTION
**Pull request by rjernst**
This commit moves the system call filtering initialization into NativeAccess. The code is essentially unmodified from its existing state, now existing within the *NativeAccess implementations.

relates #104876

### CODE REVIEW

**Review by ldematte — COMMENTED** [review]
I see we are dropping support for Solaris, FreeBSD, OpenBSD, but I suppose this is OK, as this is in line with all the rest of native functions

**Comment by rjernst** [comment]
> I see we are dropping support for Solaris, FreeBSD, OpenBSD

We never officially supported these, so I don't see this as dropping support, just cleaning up unused code. There are numerous other things that will break/fail on those platforms.

File: libs/native/src/test/java/org/elasticsearch/nativeaccess/SystemCallFilterTests.java [hunk 1]
```diff
@@ -20,15 +24,18 @@ public class SystemCallFilterTests extends ESTestCase {
     @Override
     public void setUp() throws Exception {
         super.setUp();
-        assumeTrue("requires system call filter installation", Natives.isSystemCallFilterInstalled());
+        assumeTrue(
```
[the remark this answers is not in the document]
Comment (rjernst) [reply libs/native/src/test/java/org/elasticsearch/nativeaccess/SystemCallFilterTests.java:27]: I've added several more labels that should trigger complete testing on all platforms.

[the remark this answers is not in the document]
Comment (rjernst) [reply libs/native/src/test/java/org/elasticsearch/nativeaccess/SystemCallFilterTests.java:27]: > with JDK 17 and 21

Actually, these labels won't trigger either of these versions precisely. I'll double check packaging tests, if we turn on bootstrap checks it should cover all these test scenarios.

File: libs/native/jna/src/main/java/org/elasticsearch/nativeaccess/jna/JnaKernel32Library.java [hunk 2]
```diff
@@ -98,6 +99,38 @@ public long Type() {
         }
     }
 
+    /**
+     * Basic limit information for a job object
+     *
+     * https://msdn.microsoft.com/en-us/library/windows/desktop/ms684147%28v=vs.85%29.aspx
+     */
+    public static class JnaJobObjectBasicLimitInformation extends Structure implements ByReference, JobObjectBasicLimitInformation {
```
[the remark this answers is not in the document]
Comment (rjernst) [reply libs/native/jna/src/main/java/org/elasticsearch/nativeaccess/jna/JnaKernel32Library.java:107]: Oh good catch, that entire class can be removed.

File: libs/native/src/main/java/org/elasticsearch/nativeaccess/LinuxNativeAccess.java [hunk 3]
```diff
@@ -46,4 +119,184 @@ protected void logMemoryLimitInstructions() {
...
+        // also, some of these security features get backported to old versions, checking kernel version here is a big no-no!
+        String archId = System.getProperty("os.arch");
+        final Arch arch = ARCHITECTURES.get(archId);
+        if (arch == null) {
+            throw new UnsupportedOperationException("seccomp unavailable: '" + archId + "' architecture unsupported");
+        }
+
+        // TODO: move this UOE to when binding methods in jna/jdk impls
+        // we couldn't link methods, could be some really ancient kernel (e.g. < 2.1.57) or some bug
+        /*if (linux_libc == null) {
```
[the remark this answers is not in the document]
Comment (rjernst) [reply libs/native/src/main/java/org/elasticsearch/nativeaccess/LinuxNativeAccess.java:135]: I'll do it now.

**Review by ldematte — APPROVED** [review]
As long as the additional testing/validation we discussed shows no problems (thanks for the effort!), LGTM

File: qa/ccs-rolling-upgrade-remote-cluster/build.gradle [hunk 4]
```diff
@@ -29,6 +29,7 @@ BuildParams.bwcVersions.withWireCompatible { bwcVersion, baseName ->
     versions = [bwcVersion.toString(), project.version]
     setting 'cluster.remote.node.attr', 'gateway'
     setting 'xpack.security.enabled', 'false'
+    setting 'logger.org.elasticsearch.env.NodeEnvironment', 'TRACE'
```
Comment (mark-vieira) [inline qa/ccs-rolling-upgrade-remote-cluster/build.gradle:32]: Do we want to keep this in here?

**Review by mark-vieira — APPROVED** [review]

File: qa/ccs-rolling-upgrade-remote-cluster/build.gradle (hunk 4 above)
Comment (rjernst) [reply qa/ccs-rolling-upgrade-remote-cluster/build.gradle:32]: No, i've just been adding these for debugging the windows bwc test failure.

**Pull request merged**